packages/react-components/src/components/ChatMessageComponentAsMessageBubble.tsx

```diff
@@ -0,0 +1,184 @@
+  const messageRef = useRef<HTMLDivElement | null>(null);
+  const messageActionButtonRef = useRef<HTMLElement | null>(null);
+  const [chatMessageActionFlyoutTarget, setChatMessageActionFlyoutTarget] = useState<
+    React.MutableRefObject<HTMLElement | null> | undefined
+  >(undefined);
+
+  const chatActionsEnabled = !disableEditing && message.status !== 'sending' && !!message.mine;
+  const actionMenuProps = chatMessageActionMenuProps({
+    enabled: chatActionsEnabled && allowChatActionButtonShow,
+    // Force show the action button while the flyout is open and targeting the action menu button
+    forceShow: chatMessageActionFlyoutTarget === messageActionButtonRef,
+    menuButtonRef: messageActionButtonRef,
+    onActionButtonClick: () => {
+      // Open chat action flyout, and set the context menu to target the chat message action button
+      setChatMessageActionFlyoutTarget(messageActionButtonRef);
+    },
+    theme
+  });
+
+  const longTouchPressProps = useLongPress(
+    () => {
+      // Open chat action flyout, and set the context menu to target the chat message
+      setChatMessageActionFlyoutTarget(messageRef);
+    },
+    {
+      // Don't show the action button when clicked via touch events
+      onStart: () => setAllowChatActionButtonShow(false),
+      // If the touch press didn't complete, allow the action menu to be shown on hover/focus again
+      onCancel: () => setAllowChatActionButtonShow(true),
+      captureEvent: true,
+      cancelOnMovement: true,
+      detect: LongPressDetectEvents.TOUCH
+    }
+  );
+
+  const messageContentItem = GenerateMessageContent(message, strings.liveAuthorIntro);
+
+  const chatMessage = (
+    <>
+      <div ref={messageRef} {...longTouchPressProps}>
+        <Chat.Message
+          className={mergeStyles(messageContainerStyle as IStyle)}
+          styles={messageContainerStyle}
+          content={messageContentItem}
+          author={<Text className={chatMessageDateStyle}>{message.senderDisplayName}</Text>}
+          mine={message.mine}
+          timestamp={
+            <Text data-ui-id={ids.messageTimestamp}>
+              {message.createdOn
+                ? showDate
+                  ? formatTimestampForChatMessage(message.createdOn, new Date(), strings)
+                  : formatTimeForChatMessage(message.createdOn)
+                : undefined}
+            </Text>
+          }
+          details={
+            message.editedOn ? <div className={chatMessageEditedTagStyle(theme)}>{strings.editedTag}</div> : undefined
+          }
+          positionActionMenu={false}
+          actionMenu={actionMenuProps}
+        />
+      </div>
+
+      {chatActionsEnabled && (
+        <ChatMessageActionFlyout
+          hidden={!chatMessageActionFlyoutTarget}
+          target={chatMessageActionFlyoutTarget ?? undefined}
+          onDismiss={() => {
+            setChatMessageActionFlyoutTarget(undefined);
```

Comment:
Please add a comment here similar to lines 72 and 74.

Reply:
updated